### DISCUSSION TITLE
How well does this scale with high thread count cpus?

### DISCUSSION THREAD

**Discussion by Zerophase** [Chia-Network/chia-blockchain#4248] [Support]
I'm thinking about using raid zero with a shit ton of disks, and want at least 1 thread per plot. So, I'm thinking about 128 logical cores. I'm going to have about 44 Tibs on nvme drives for plotting, and a bunch of disks for storage. Are there any benefits for going beyond 1 thread per plot, at any of the phases?

I'm basically trying to figure out where between 48 logical cores and 128 logical cores adding more threads would not improve performance significantly. I'm guessing there might even be a point where adding more threads decreases performance. What's the trade off between individual core speed, and thread counts?

System will most likely have 128 gigs of ram too with a slight overclock above xmp, in case that impacts the benefit of having more threads.

**Comment by cccat6** [comment]
Multithread can only apply in phase1. The remaining 3 phases can only use a single thread.
In phase1, 2 threads are significantly better than 1 thread. More than 4 thread does not have significant improvement.
If you have 44TiB (please make sure it's bit or byte), ideally you can parallel more than 200 tasks at the same time.
I don't really know how can you apply that much nvme drive on a single computer with low CPU and low RAM. But obviously, your plotting machine is not a good configuration.

**Reply by Zerophase** [reply]
Well, sure I'll probably drop from a $4,000k dollar cpu to a $2k one. I definitely want at least 48 logical cores. I can go up to 256 gigs of ram eventually, maybe more depending on what my board can support. I do have 128 gigs of Samsung B-die already. The only limitations with the current planned build is it creates plots way too fast, and I need to buy a ton of HDDs per month to move to my farming machine. That might make it unprofitable, depending on how farming compares to plotting. 

So, I need at least 4 threads free for phase 1 for each plot I'm creating, and have enough threads for staggering the startup of different plots? What ghz threshold should I limit the cpu to for viability? 4.5ghz to 5ghz across all cores as the minimum?

Does frequency or latency impact performance more from ram?

Does random reads and writes matter more, or does sequential have more impact? 

Other than that I'll be playing around with Gentoo trying to squeeze more performance out. I have some settings from my main Arch rig I'll probably move over there to help with performance, on top of the optimizations Gentoo can turn on. Has anyone tested the CK kernel vs the stock Linux kernel?

**Reply by cccat6** [reply]
There should be no CPU that can reach 4.5GHz with 48 logical cores. What you are talking about seems Xeon, Threadripper, or EPYC.
You don't necessarily give all tasks 4 threads, 2 is ok and 4 is good as well. There is no strict restriction on that.
You do need to use staging in order to fully use your CPU but not overload it. You need to do the test and figure out how much time it should delay.
RAM latency and frequency do affect, but hard to tell how much it will be affected. It usually not the point that needs to take care of.
Random RW and sequential RW are both crucial. For plotting, random RW usually has more impact. That's why we are using SSD but not HDD. Even HDD raid0 will not have much improvement.
I guess you just thinking too much and really ideally. You seem not clear what does plotting needs. In my practice, multiple small PCs are much better than a single huge server in the same money spent.
If you are looking for professional Chia mining, you should stop and ask the expert for help. If you do that just for fun. Go ahead, you should not care what you currently have, because what you planning to do and the strange hardware configuration is really make no sense for me.

**Reply by Zerophase** [reply]
>There should be no CPU that can reach 4.5GHz with 48 logical cores. What you are talking about seems Xeon, Threadripper, or EPYC
.
The 3960x can hit 4.5ghz on all 48 logical cores. 24 hardware cores, and I don't know the AMD term for hyperthreading. But, logical cores is the concept. Usually I find it's possible to push systems up to 1.5 times their logical core count for finishing a task fast. Makes the system unresponsive for a user, though. I'm looking at thread rippers as they're the best cpus currently. I'm looking at thread ripper for the cpu.

> You don't necessarily give all tasks 4 threads, 2 is ok and 4 is good as well. There is no strict restriction on that.
You do need to use staging in order to fully use your CPU but not overload it. You need to do the test and figure out how much time it should delay.

I can set 4 for phase 1 in the program, and it'll automatically do that right. I'm sure there's a way to stagger everything so, I have multiple tasks in phase 1, while others are starting. Thus keeping my cpu busy.

> Random RW and sequential RW are both crucial. For plotting, random RW usually has more impact. That's why we are using SSD but not HDD. Even HDD raid0 will not have much improvement.

I'm just getting HDDs for storing plots long term for farming.

>If you are looking for professional Chia mining, you should stop and ask the expert for help. If you do that just for fun. Go ahead, you should not care what you currently have, because what you planning to do and the strange hardware configuration is really make no sense for me.

That's why I'm asking on their Github forum. The configuration is based around what the computer does, and nvme drives can push crazy data rates a second. Should be faster than ram plotting on ram disks from what I've seen, and ram above 256 gigs is prohibitively expensive. 

I'm trying to write as much data as possible per second. So, I mint at least a coin a day.

**Comment by BrandtH22** [comment]
If you are still experiencing the issues it is recommended to upgrade to the latest version of chia. Please download it from here: https://www.chia.net/downloads/

Since this issue has been open for a number of years without additional comments, we will be closing this ticket but if we have closed this ticket in error do not hesitate to reach out to us again with any followup questions, comments, or if the issue persists after an update.

```
Please note that upgrading major versions of chia generally requires the config.yaml to be recreated using these steps:
1. Stop chia
2. Rename the config or move it to a different directory
3. In CLI run chia init
4. Copy any custom settings from the old config to the new (plot drive directories, pool info, trusted node info, etc) - be careful here as yaml is very picky about formatting including those leading spaces
5. Start Chia
```

The best place to reach our support team is on Discord (https://discord.gg/chia) or by reopening this ticket.
